### PR TITLE
GCATI-273more, re-instate the cursed Detail model, after corrections

### PR DESCRIPTION
**Pull request by decet**
See throwaway uam-apis branch of same name for gen.sh modified for codegen.
See attached screenshot of successful codegen run.
Screen Shot 2020-02-24 at 3 08 16 PM

### CODE REVIEW

File: datacollection/openapi/v3/schema/adsb.yaml [hunk 1]
```diff
@@ -61,35 +56,307 @@ components:
...
+          uatMopsVersion:
+            type: integer
+            enum: [0,1]
+            description: 1 = DO-282A, 2 = DO-282B
+          callSignID:
+            type: integer
+            enum: [0,1]
+            description: 0 = Fightplan, 1 = CallSign
+
+  #    RID:
```
Comment (issmith1) [inline datacollection/openapi/v3/schema/adsb.yaml:322]: what is this

File: datacollection/openapi/v3/schema/adsb.yaml [hunk 2]
```diff
@@ -61,35 +56,307 @@ components:
...
+            type: integer
+            description: Phone number attached to the mission.
+              todo sic. Cannot possibly be right. long or string?
+          fltplanFiled:
+            type: integer
+            enum: [0,1]
+            description: NOTAM/UOA flight plan filed flag
+              todo, not checked, no example
 
+# timestamp stolen from .. todo: should use commons
```
Comment (issmith1) [inline datacollection/openapi/v3/schema/adsb.yaml]: timestamp is hardest part of codegen.  What should this be

File: datacollection/openapi/v2/schema/schema-uam-adsb.yaml [hunk 3]
```diff
@@ -1,366 +0,0 @@
...
-          description: Squawk code
-        altitudeType:
-          type: integer
-          enum: [0,1]
-          description: Altitude Source, 0 is pressure, 1 is geometric
-        callsign:
-          type: string
-          pattern: '^[A-Z0-9]+ *$'
-        emitterType:
-          type: number
```
Comment (issmith1) [inline datacollection/openapi/v2/schema/schema-uam-adsb.yaml]: I found a error in codegen where the enum was type=number but needed to be integer.    In any case we can be consistent by changing all enums to integer

File: datacollection/openapi/v3/schema/adsb.yaml (hunk 1 above)
Comment (decet) [reply datacollection/openapi/v3/schema/adsb.yaml:322]: That looks like a typographical error: should be Flightplan. I'll fix it.

File: datacollection/openapi/v3/schema/adsb.yaml (hunk 2 above)
Comment (decet) [reply datacollection/openapi/v3/schema/adsb.yaml]: It's a date/time string formatted according to ISO8601, as referenced by https://tools.ietf.org/html/rfc3339#section-5.6, as referenced by https://swagger.io/docs/specification/data-models/data-types/

I don't think I know enough about the intended downstream users to specify a particular approach in codegen. I'm trying to provide as much information for JSON validation to the endpoint as I can.

I can remove the comment, and I'm not convinced that we should tie this to our published commons. uAvionics could change it (although I doubt they will).

File: datacollection/openapi/v3/schema/adsb.yaml [hunk 4]
```diff
@@ -61,34 +56,305 @@ components:
             description: Squawk code
```
Comment (decet) [inline datacollection/openapi/v3/schema/adsb.yaml:57]: That's a pattern I've got there on that number. I'll fix it.

**Pull request merged**